lib/experimental/fonts/font-library/class-wp-rest-font-library-controller.php

```diff
@@ -67,6 +67,69 @@ public function register_routes() {
 				),
 			)
 		);
+
+		register_rest_route(
+			$this->namespace,
+			'/' . $this->rest_base . '/collections',
+			array(
+				array(
+					'methods'             => WP_REST_Server::READABLE,
+					'callback'            => array( $this, 'get_font_collections' ),
+					'permission_callback' => array( $this, 'update_font_library_permissions_check' ),
+				),
+			)
+		);
+
+		register_rest_route(
+			$this->namespace,
+			'/' . $this->rest_base . '/collections' . '/(?P<id>[\/\w-]+)',
+			array(
+				array(
+					'methods'             => WP_REST_Server::READABLE,
+					'callback'            => array( $this, 'get_font_collection' ),
+					'permission_callback' => array( $this, 'update_font_library_permissions_check' ),
+				),
+			)
+		);
+	}
+
+	/**
+	 * Gets a font collection.
+	 *
+	 * @since 6.4.0
+	 *
+	 * @param WP_REST_Request $request Full details about the request.
+	 * @return WP_REST_Response|WP_Error Response object on success, or WP_Error object on failure.
+	 */
+	public function get_font_collection( $request ) {
+		$id         = $request->get_param( 'id' );
+		$collection = WP_Font_Library::get_font_collection( $id );
+
+		if ( is_wp_error( $collection ) ) {
+			return new WP_Error(
+				'font_collection_not_found',
+				__( 'Font collection not found.', 'gutenberg' ),
+				array( 'status' => 404 )
+			);
```

Comment:
If an instance of `WP_Error` is returned, why not return that instance instead of creating a new one?

The `array( 'status' => 404 )` can be added to `WP_Error` in  `WP_Font_Library::get_font_collection()`. And then it can be returned.

```suggestion
			return $collection;
```